subhub/customer.py

```diff
@@ -65,27 +65,38 @@ def create_customer(
 
 
 def existing_or_new_customer(
-    subhub_accouunt: SubHubAccount,
+    subhub_account: SubHubAccount,
     user_id: str,
     email: str,
     source_token: str,
     origin_system: str,
     display_name: str,
 ) -> stripe.Customer:
-    db_account = subhub_accouunt.get_user(user_id)
-    if not db_account:
+    customer = fetch_customer(subhub_account, user_id)
+    if not customer:
         return create_customer(
-            subhub_accouunt, user_id, email, source_token, origin_system, display_name
+            subhub_account, user_id, email, source_token, origin_system, display_name
         )
-    customer_id = db_account.cust_id
-    return existing_payment_source(customer_id, source_token)
+    return existing_payment_source(customer, source_token)
+
+
+def fetch_customer(subhub_account: SubHubAccount, user_id: str) -> stripe.Customer:
+    customer = None
+    db_account = subhub_account.get_user(user_id)
+    if db_account:
+        customer = stripe.Customer.retrieve(db_account.cust_id)
+        if "deleted" in customer and customer["deleted"]:
+            subhub_account.remove_from_db(user_id)
```

Comment:
This needs to be updated once Marty's "soft-delete" work has been completed and merged into master

Reply:
Soft delete is complete! @jackiemunroe